client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx

```diff
@@ -67,48 +71,91 @@ const PendingGappsTosNotice = React.createClass( {
 
 		switch ( severity ) {
 			case 'warning':
-				return this.translate( 'Attention!', translationOptions );
+				return this.props.translate( 'Attention!', translationOptions );
 
 			case 'error':
-				return this.translate( 'Urgent!', translationOptions );
+				return this.props.translate( 'Urgent!', translationOptions );
 
 			default:
-				return this.translate( 'You\'re almost there!', translationOptions );
+				return this.props.translate( 'You\'re almost there!', translationOptions );
 		}
 	},
 
 	generateLogInClickHandler( { domainName, user, severity, isMultipleDomains } ) {
 		return () => {
-			this.recordEvent( 'pendingAccountLogInClick', { siteSlug: this.props.siteSlug, domainName, user, severity, isMultipleDomains, section: this.props.section } );
+			this.recordEvent( 'pendingAccountLogInClick', {
+				domainName,
+				isMultipleDomains,
+				severity,
+				user,
+				section: this.props.section,
+				siteSlug: this.props.siteSlug
+			} );
+		};
+	},
+
+	generateFixClickHandler() {
+		return () => {
+			this.recordEvent( 'fixPendingEmailSiteNoticeClick', this.props.siteSlug );
 		};
 	},
 
+	compactNotice() {
+		const severity = this.getNoticeSeverity(),
+			href = this.props.domains.length === 1
+				? domainManagementEmail( this.props.siteSlug, this.props.domains[ 0 ].name )
+				: domainManagementEmail( this.props.siteSlug );
+
+		return (
+			<Notice
+				isCompact={ this.props.isCompact }
```

Comment:
Yes - it's the "React" way ;) But, seriously, while you are correct in the sense that `this.props.isCompact` will always be `true` for this code, this might change in the future (refactoring, etc.). I'd rather not hardcode such values, when I can reliably depend on a property of the component. It doesn't hurt the readability of the code, either, IMHO - it seems reasonable that the `isCompact` property of the parent component controls the `isCompact` property of the child notice.